octopus/src/main/scala/octopus/dsl.scala

```diff
@@ -58,118 +58,119 @@ object dsl {
     def ruleOption(pred: T => Option[Boolean], whenInvalid: String, whenNone: String): Validator[T] =
       compose(ValidationRules.ruleOption(pred, whenInvalid, whenNone))
 
-
-    def async: AsyncValidatorAsyncOps[T] =
-      new AsyncValidatorAsyncOps[T](AsyncValidator.lift(v))
+    def async[F[_]: App]: AsyncValidatorAsyncOps[F, T] =
+      new AsyncValidatorAsyncOps[F, T](AsyncValidator.lift[F, T](v))
   }
 
-  class AsyncValidatorAsyncOps[T](val v: AsyncValidator[T]) extends AnyVal {
+  class AsyncValidatorAsyncOps[M[_]: App, T](val v: AsyncValidator[M, T]) {
```

Comment:
Dropping `extends AnyVal` forces compiler to generate additional boxing code on every dsl operation. This is not nice. Can we somehow bring it back? Either by requiring implicit instance from method level, or just maybe encapsulating that inside async validator, if possible at all?

Reply:
`extends AnyVal` is back there